### PR TITLE
Vue lern store

### PR DESCRIPTION
**Pull request by j-jmnz**
# Issue: [VUE-130](https://ticketsystem.schul-cloud.org/browse/VUE-130)

## Description

- Changed Content page pagination to infinite scrolling
- Changed ContentCard styling

## Screenshot
Screenshot from 2019-11-12 14-31-47

### CODE REVIEW

**Comment by janrenz** [comment]
please resolve conflicts

File: src/components/base/BaseLink.vue [hunk 1]
```diff
@@ -142,7 +144,9 @@ export default {
 	color: var(--color-primary);
 	text-decoration: none;
 	cursor: pointer;
-	border-bottom: 2px solid var(--color-gray);
+	&.underlined {
+		border-bottom: 2px solid var(--color-gray);
+	}
```
Comment (adrianjost) [inline src/components/base/BaseLink.vue]: - [ ] Links should be underlined by default. If you want your links to have no special style, please use the `noStyles: true` prop.
=> Please revert the changes made here

File: src/components/molecules/ContentCard.vue [hunk 2]
```diff
@@ -0,0 +1,138 @@
+<template>
+	<BaseCard v-bind="$props">
+		<div class="content-card">
+			<template v:slot:content>
+				<div class="content">
+					<BaseLink :href="url" target="_blank">
```
Comment (adrianjost) [inline src/components/molecules/ContentCard.vue]:
```suggestion
					<BaseLink :href="url" target="_blank" :no-style="true">
```
add the `noStyle`-prop here that I explained above to remove the underline

File: src/components/molecules/ContentCard.vue [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+						</span>
+					</div>
+					<div class="content__description">{{ description }}</div>
+				</div>
+			</template>
+			<template v:slot:footer>
+				<div class="footer">
+					<div class="footer__melden">
+						<a
+							href="mailto:inhalte@schul-cloud.org?Subject=Melden%20des%20Inhaltes%20mit%20der%20ID%20&body=Liebes%20%20Team%2C%0Ahiermit%20m%C3%B6chte%20ich%20den%20im%20Betreff%20genannten%20Inhalt%20melden%2C%20da%3A%0A%5Bhier%20bitte%20Ihre%20Gr%C3%BCnde%5D"
```
Comment (adrianjost) [inline src/components/molecules/ContentCard.vue]: could you please put this string into a computed property and generate the uri encoded version from a regular string? This would improve readability and changeability a lot.
Best case would be to use the language files for this string as well.

Something like this:
```suggestion
							:href="`mailto:inhalte@schul-cloud.org?${mailContent}`"
```

```vue
computed: {
  reportMail(){
    const mailContent = {
      Subject: this.$t("contentcard.report.subject"),
      body: this.$t("contentcard.report.body"),
    }
    return Object.keys(mailContent).map(key => key + '=' + encodeURIComponent(params[key])).join('&');
  }
}
```
add your language keys (`contentcard.report.subject`) to `locales/de.json` and use [our naming scheme for these keys](https://vuepress.test.schul-cloud.org/1-Tutorials/Localization.html#verwendung) (sorry, this part of the documentation is currenly only available in german)

File: src/components/molecules/ContentEmptyState.vue [hunk 4]
```diff
@@ -0,0 +1,59 @@
+<template>
+	<div class="empty-state-container">
+		<img
+			class="empty-state-container__image"
+			:src="getRandomSVG"
```
Comment (adrianjost) [inline src/components/molecules/ContentEmptyState.vue]: @dominikbruechner is it intended to use a random empty state image? I like the idea.

File: src/store/content.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+		commit("setLoadingTrue")
+		const query = payload || {};
+		query["$limit"] = 9;
+		const res = await this.$axios.$get("/content/search", {
+			params: query,
+		});
+		commit("setResources", res);
+		commit("setLoadingFalse")
+	},
+	async addResources({
```
Comment (adrianjost) [inline src/store/content.js]: What is `addResources` doing? From the name I would guess it creates some kind of resource but it is only doing a GET request? Maybe the name is missleading?

File: src/store/content.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
+export const actions = {
+	async getResources({
+		commit
+	}, payload = {}) {
+		commit("setLoadingTrue")
+		const query = payload || {};
+		query["$limit"] = 9;
```
Comment (adrianjost) [inline src/store/content.js]: I think it would be better to make the limit attribute also overrideable. You can use Object.assign() or the spread syntax for that.

```
const query = Object.assign({$limit: 9}, payload || {})
```

File: src/store/content.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
+export const actions = {
+	async getResources({
+		commit
+	}, payload = {}) {
+		commit("setLoadingTrue")
```
Comment (adrianjost) [inline src/store/content.js]:
```suggestion
		commit("setLoading", true)
```
You could write a generic method (mutation) for that, that accepts a parameter
https://vuex.vuejs.org/guide/mutations.html#commit-with-payload

File: src/store/content.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+		const query = payload || {};
+		query["$limit"] = 9;
+		const res = await this.$axios.$get("/content/search", {
+			params: query,
+		});
+		commit("setResources", res);
+		commit("setLoadingFalse")
```
Comment (adrianjost) [inline src/store/content.js]:
```suggestion
		commit("setLoading", false)
```

File: src/store/content.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+	},
+	async addResources({
+		commit
+	}, payload = {}) {
+		commit("setLoadingTrue")
```
Comment (adrianjost) [inline src/store/content.js]:
```suggestion
		commit("setLoading", true)
```

File: src/store/content.js [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+		const query = payload || {};
+		const res = await this.$axios.$get("/content/search", {
+			params: query,
+		});
+		commit("addResources", res);
+		commit("setLoadingFalse")
```
Comment (adrianjost) [inline src/store/content.js]:
```suggestion
		commit("setLoading", false)
```

File: stories/base.stories.js [hunk 11]
```diff
@@ -220,9 +213,9 @@ storiesOf("Base|Base UI", module)
 		components: { BaseLink },
 		template: `
 			<div>
-				<base-link href="https://schul-cloud.org">external Link to https://schul-cloud.org</base-link>
-				<base-link to="/news">Internal Link to /news</base-link>
-				<base-link name="news">Internal Link with name "news"</base-link>
+				<base-link href="https://schul-cloud.org" underlined>external Link to https://schul-cloud.org</base-link>
+				<base-link to="/news" underlined>Internal Link to /news</base-link>
+				<base-link name="news" underlined>Internal Link with name "news"</base-link>
```
Comment (adrianjost) [inline stories/base.stories.js]: as explained above, the underline changes should be reverted. 
A usefull git command to revert files to the state of another branch is:
`git checkout origin/develop stories/base.stories.js`

File: stories/Cards.stories.js [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+				:picture="picture"
+				:eventDate="eventDate"
+				:isLandscape="isLandscape"
+				:color="color"
+			>
+				{{content}}
+			</news-card>
+		</div>
+		`,
+	}));
```
Comment (adrianjost) [inline stories/Cards.stories.js]: - [ ] `NewsCard` is not part of this pull request, so it shouldn't be added yet. please remove it from here.

File: stories/Cards.stories.js [hunk 13]
```diff
@@ -0,0 +1,79 @@
+import { storiesOf } from "@storybook/vue";
+import { text, number, color, boolean } from "@storybook/addon-knobs";
+
+import CourseCard from "@components/molecules/CourseCard";
+import notes from "@docs/storybook/courseCard.md";
+import NewsCard from "@components/molecules/NewsCard";
```
Comment (adrianjost) [inline stories/Cards.stories.js]:
```suggestion
```

File: src/mixins/multiline-ellipsis.scss [hunk 14]
```diff
@@ -0,0 +1,45 @@
+@import "@styles";
```
Comment (adrianjost) [inline src/mixins/multiline-ellipsis.scss]: This is a duplicate of `src/utils/multiline-ellipsis.scss`. Please remove this file.

File: src/components/molecules/CourseCard/index.unit.js [hunk 15]
```diff
@@ -50,21 +46,16 @@ describe("@components/molecules/CourseCard", () => {
...
 		});
-		expect(wrapper.find("base-icon-stub").exists()).toBe(true);
-		expect(wrapper.find(".assignments-label").text()).toBe(
-			course.newAssignments.toString()
-		);
-		expect(wrapper.find(".notification-dot").text()).toBe(
-			course.notification.toString()
-		);
+		expect(wrapper.text()).toContain(course.newAssignments.toString());
+		expect(wrapper.text()).toContain(course.notification.toString());
```
Comment (adrianjost) [inline src/components/molecules/CourseCard/index.unit.js]: - [ ] Why have you changes something in this file? CourseCards were not changed in this pull? Please revert these changes.

File: src/mixins/infiniteScrolling.js [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+		});
+	},
+	beforeDestroy() {
+		window.removeEventListener("scroll", () => {
+			this.bottom = this.bottomVisible();
+			this.scrollY = window.scrollY;
+		});
+	},
+	methods: {
+		bottomVisible() {
```
Comment (adrianjost) [inline src/mixins/infiniteScrolling.js]: Please use our naming guidelines for mixins to prevent future namespace conflicts:
https://vuepress.test.schul-cloud.org/0-GettingStarted/3-DirectoryStructure.html#src-mixins
```suggestion
		$_bottomVisible() {
```

File: src/mixins/infiniteScrolling.js [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+			const {
+				scrollY
+			} = window;
+			const visible = document.documentElement.clientHeight;
+			const pageHeight = document.documentElement.scrollHeight;
+			const bottomOfPage = visible + scrollY >= pageHeight - 2;
+			return bottomOfPage || pageHeight < visible;
+		},
+		backToTop() {
```
Comment (adrianjost) [inline src/mixins/infiniteScrolling.js]:
```suggestion
		$_backToTop() {
```

File: src/mixins/infiniteScrolling.js [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+	data() {
+		return {
+			bottom: false,
+			scrollY: 0,
+		}
+	},
+	created() {
+		window.scrollTo(0, 0);
+		window.addEventListener("scroll", () => {
+			this.bottom = this.bottomVisible();
```
Comment (adrianjost) [inline src/mixins/infiniteScrolling.js]:
```suggestion
			this.bottom = this.$_bottomVisible();
```

File: src/mixins/infiniteScrolling.js [hunk 19]
```diff
@@ -0,0 +1,37 @@
...
+			this.scrollY = window.scrollY;
+		});
+	},
+	beforeDestroy() {
+		window.removeEventListener("scroll", () => {
+			this.bottom = this.bottomVisible();
```
Comment (adrianjost) [inline src/mixins/infiniteScrolling.js]:
```suggestion
			this.bottom = this.$_bottomVisible();
```

File: src/pages/content/index.vue [hunk 20]
```diff
@@ -1,72 +1,104 @@
...
 export default {
 	components: {
 		Searchbar,
-		Pagination,
+		ContentCard,
+		ContentEmptyState,
+	},
+	mixins: [infiniteScrolling],
+	async asyncData({ store }) {
+		return Promise.all([store.dispatch("content/getResources")]);
```
Comment (adrianjost) [inline src/pages/content/index.vue]:
```suggestion
		return store.dispatch("content/getResources");
```
There is only one Promise you are waiting for so you can return this promise directly.

File: src/store/content.js (hunk 5 above)
Comment (j-jmnz) [reply src/store/content.js]: `addResources` gets called every time the scroller gets to the bottom of the page, adding resources to the state, hence the name. In my opinion the name is descriptive in relation to its functionality, but if I'm mistaken please let me know.

File: src/components/molecules/ContentCard.unit.js [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+				description: "Description",
+			},
+		});
+		expect(wrapper.find(".content__title").text()).toBe("Title");
+		expect(wrapper.find(".footer__info").text()).toBe("via ProviderName");
+		expect(wrapper.find(".content__thumbnail").attributes().src).toBe(
+			"Thumbnail"
+		);
+		expect(wrapper.find(".content__tags-tag").text()).toBe("Tag");
+		expect(wrapper.find(".content__description").text()).toBe("Description");
```
Comment (adrianjost) [inline src/components/molecules/ContentCard.unit.js]: great tests 💯 👍

File: src/components/molecules/ContentCard.vue [hunk 22]
```diff
@@ -0,0 +1,144 @@
+<template>
+	<base-card v-bind="$props">
```
Comment (adrianjost) [inline src/components/molecules/ContentCard.vue]: Are you shure you mean `$props` and not `$attrs`. `$props` contain all the prop values you have defined in this component,`$attrs` contains everything else passed down to this component.

File: src/components/molecules/ContentCard.vue [hunk 23]
```diff
@@ -0,0 +1,144 @@
...
+		originId: { type: String, default: "" },
+		providerName: { type: String, default: "" },
+		tags: { type: Array, default: () => [] },
+		thumbnail: { type: String, default: "" },
+		title: { type: String, default: "" },
+		url: { type: String, default: "" },
+	},
+	computed: {
+		mailContent() {
+			return "mailto:inhalte@schul-cloud.org?Subject=Melden%20des%20Inhaltes%20mit%20der%20ID%20&body=Liebes%20%20Team%2C%0Ahiermit%20m%C3%B6chte%20ich%20den%20im%20Betreff%20genannten%20Inhalt%20melden%2C%20da%3A%0A%5Bhier%20bitte%20Ihre%20Gr%C3%BCnde%5D";
```
Comment (adrianjost) [inline src/components/molecules/ContentCard.vue]: - [ ] Please make this more readable and maintainable by using the language files

It could work somehow like this:
```js
computed: {
  reportMail(){
    const mailContent = {
      Subject: this.$t("contentcard.report.subject"), // get contentn from language file with key `contentcard.report.subject`
      body: this.$t("contentcard.report.body"),
    }
    // encode all the texts to form a valid url
    const querystring = Object.keys(mailContent).map(key => key + '=' + encodeURIComponent(params[key])).join('&');
    const email = this.$t("contentcard.report.email);
    // form email url
    return `mailto:${email}?${querystring}`
  }
}
```
If you need help, don't hesitate to contact me.

File: src/components/molecules/ContentEmptyState.vue [hunk 24]
```diff
@@ -0,0 +1,53 @@
...
+			alt="empty-state-img"
+		/>
+		<h3 class="empty-state-container__title">Oops</h3>
+		<p class="empty-state-container__message"
+			>We couldn't find any search result, please try entering other keywords</p
```
Comment (adrianjost) [inline src/components/molecules/ContentEmptyState.vue]: - [ ] Please use the lanugage files for all strings

Our docs: [https://vuepress.test.schul-cloud.org/1-Tutorials/Localization.html#verwendung](http://translate.google.com/translate?sl=de&tl=en&u=https%3A%2F%2Fvuepress.test.schul-cloud.org%2F1-Tutorials%2FLocalization.html%23verwendung)
i18n docs: https://kazupon.github.io/vue-i18n/started.html#javascript

File: src/pages/content/index.vue [hunk 25]
```diff
@@ -1,72 +1,109 @@
...
+		<div v-if="scrollY > 115" class="content__back-to-top">
+			<p @click="$_backToTop">
+				<i class="icon fa fa-chevron-circle-up"></i>
+			</p>
+		</div>
+		<div class="content">
+			<searchbar
+				v-model.lazy="searchQuery"
+				class="content__searchbar"
+				placeholder="Suche nach..."
```
Comment (adrianjost) [inline src/pages/content/index.vue]: - [ ] Please use the language files here as well
```suggestion
				:placeholder="$t('pages.content.index.search_for')"
```

File: src/pages/content/index.vue [hunk 26]
```diff
@@ -1,72 +1,109 @@
...
+				:loading="loading"
+			/>
+			<p class="content__total">
+				<span v-if="searchQuery.length > 0">
+					{{ resources.total }} Suchergebnisse für "{{ searchQuery }}"
```
Comment (adrianjost) [inline src/pages/content/index.vue]: - [ ] Please use the language files here as well

File: src/pages/content/index.vue [hunk 27]
```diff
@@ -1,72 +1,109 @@
...
+				</span>
+				<span v-else> {{ resources.total }} resources </span>
```
Comment (adrianjost) [inline src/pages/content/index.vue]: - [ ] Please use the language files here as well
You can pass variables to the lanugage files and pluralize:
https://kazupon.github.io/vue-i18n/guide/pluralization.html

```suggestion
				<span v-else> $t('pages.content.index.found_resources', { resources: resources.total }) </span>
```
and in the language file:
```
"pages.content.index.found_resources": 'keine Ressourcen gefunden | ein Ergebniss gefunden | {resources} Ergebnisse gefunden'
```

File: src/static/empty-state/emptyStateSvg_lehrer1.svg [hunk 28]
```diff
@@ -0,0 +1 @@
...583)" d="M112.04 55.6h3.23v1.95h-3.23z"/><path class="cls-20" d="M139.39 49.45v26.08h-1.25a3.15 3.15 0 01-3.14-3.14V53.85a4.39 4.39 0 014.39-4.39z" transform="rotate(112.49 137.196 62.496)"/><path d="M133.42 53.26a4.39 4.39 0 014.39 4.39V75a4.39 4.39 0 01-4.39 4.39V53.26z" transform="rotate(112.49 135.618 66.309)" fill="#365767"/><path class="cls-19" d="M125.75 60.13l-1.45-.6 1.21-2.9a1.25 1.25 0 011.63-.68l.29.12z"/><path d="M148.21 64.68l.26.11a1.73 1.73 0 01.94 2.27l-1 2.45-1.87-.77z" fill="#385969"/><path d="M124.17 63.94l-.29-.12a1.25 1.25 0 01-.68-1.63l1.21-2.9 1.44.59zm22.47 4.55l1.86.77-1 2.45a1.74 1.74 0 01-2.28.94l-.26-.1z" fill="#6d7f8c"/><path class="cls-20" d="M88.32 65.54A19.61 19.61 0 1077.7 39.93a19.6 19.6 0 0010.62 25.61zm1.74-4.2a15.06 15.06 0 1119.67-8.15 15.05 15.05 0 01-19.67 8.15z"/><path d="M77.7 39.94l4.2 1.74a15.06 15.06 0 0027.82 11.52l4.2 1.74a19.6 19.6 0 01-36.22-15z" fill="#1f4a5a"/><path d="M109.79 53.16L82 41.64a15.06 15.06 0 0127.82 11.52z" fill="#fff" opacity=".34"/><path class="cls-16" d="M140.93 50.9l-9.08-2.1c-3.08-.71-7.83 1.86-8.74 5.82l-1 6a7.65 7.65 0 005.2 8.68l8.89 2.06c4.4 1 9-2.74 10.33-8.39 1.3-5.65-1.21-11.05-5.6-12.07z"/></g></g></svg>
```
Comment (adrianjost) [inline src/static/empty-state/emptyStateSvg_lehrer1.svg]: - [ ] Please rename the svgs to be more declerative. Something like: `emptyStateSvg_teacher_male_hand_raised.svg`

File: src/static/empty-state/emptyStateSvg_lehrer2.svg [hunk 29]
```diff
@@ -0,0 +1 @@
...6.584)" d="M143.92 55.6h3.23v1.95h-3.23z"/><path class="cls-20" d="M171.27 49.45v26.08H170a3.15 3.15 0 01-3.15-3.15V53.85a4.39 4.39 0 014.39-4.39z" transform="rotate(112.49 169.072 62.497)"/><path d="M165.3 53.26a4.39 4.39 0 014.39 4.39V75a4.39 4.39 0 01-4.39 4.39V53.26z" transform="rotate(112.49 167.493 66.31)" fill="#365767"/><path class="cls-19" d="M157.63 60.13l-1.45-.6 1.21-2.9A1.25 1.25 0 01159 56l.29.12z"/><path d="M180.09 64.68l.26.11a1.73 1.73 0 01.94 2.27l-1 2.45-1.87-.77z" fill="#385969"/><path d="M156.05 63.94l-.29-.12a1.25 1.25 0 01-.68-1.63l1.21-2.9 1.44.59zm22.46 4.55l1.87.77-1 2.45a1.74 1.74 0 01-2.28.94l-.26-.1z" fill="#6d7f8c"/><path class="cls-20" d="M120.2 65.54a19.61 19.61 0 10-10.62-25.61 19.6 19.6 0 0010.62 25.61zm1.74-4.2a15.06 15.06 0 1119.67-8.15 15 15 0 01-19.67 8.15z"/><path d="M109.57 39.94l4.21 1.74A15.06 15.06 0 00141.6 53.2l4.2 1.74a19.61 19.61 0 01-36.23-15z" fill="#1f4a5a"/><path d="M141.67 53.16l-27.82-11.52a15.06 15.06 0 0127.82 11.52z" fill="#fff" opacity=".34"/><path class="cls-16" d="M172.81 50.9l-9.09-2.1c-3.07-.71-7.82 1.86-8.73 5.82l-1 6a7.65 7.65 0 005.2 8.68l8.89 2.06c4.4 1 9-2.74 10.33-8.39 1.3-5.65-1.21-11.05-5.6-12.07z"/></g></g></svg>
```
Comment (adrianjost) [inline src/static/empty-state/emptyStateSvg_lehrer2.svg]: - [ ] Please rename the svgs to be more declerative. Something like: `emptyStateSvg_teacher_male.svg`

File: src/pages/content/index.vue [hunk 30]
```diff
@@ -1,72 +1,109 @@
...
-			<div v-for="content of searchResults" :key="content._id" class="column">
-				<pre>
-					{{ content }}
-				</pre
-				>
+	<section>
+		<div v-if="scrollY > 115" class="content__back-to-top">
+			<p @click="$_backToTop">
+				<i class="icon fa fa-chevron-circle-up"></i>
+			</p>
```
Comment (adrianjost) [inline src/pages/content/index.vue]: Please use the FAB here: `<floating-fab icon="add" @click="$_backToTop" />`

File: src/mixins/infiniteScrolling.js [hunk 31]
```diff
@@ -0,0 +1,35 @@
...
+	methods: {
+		$_bottomVisible() {
+			const { scrollY } = window;
+			const visible = document.documentElement.clientHeight;
+			const pageHeight = document.documentElement.scrollHeight;
+			const bottomOfPage = visible + scrollY >= pageHeight - 2;
+			return bottomOfPage || pageHeight < visible;
+		},
+		$_backToTop() {
+			window.scrollTo(0, 0);
```
Comment (adrianjost) [inline src/mixins/infiniteScrolling.js]: - [ ] You can use smoothScroll to make the experience even better: `window.scrollTo({ top: 0, behavior: 'smooth' })`
https://stackoverflow.com/questions/42261524/how-to-window-scrollto-with-a-smooth-effect

File: src/pages/content/index.vue [hunk 32]
```diff
@@ -1,72 +1,109 @@
...
 
 export default {
 	components: {
 		Searchbar,
-		Pagination,
+		ContentCard,
+		ContentEmptyState,
+		BaseGrid,
+	},
+	mixins: [infiniteScrolling],
```
Comment (adrianjost) [inline src/pages/content/index.vue]: - [ ] please use the full width layout
```suggestion
	mixins: [infiniteScrolling],
	layout: "loggedInFull",
```

File: locale/de.json [hunk 33]
```diff
@@ -1,4 +1,7 @@
 {
+	"contentcard.report.email": "inhalte@schul-cloud.org",
+	"contentcard.report.body": "Melden%20des%20Inhaltes%20mit%20der%20ID%20&",
+	"contentcard.report.subject": "Liebes%20%20Team%2C%0Ahiermit%20m%C3%B6chte%20ich%20den%20im%20Betreff%20genannten%20Inhalt%20melden%2C%20da%3A%0A%5Bhier%20bitte%20Ihre%20Gr%C3%BCnde%5D",
```
Comment (adrianjost) [inline locale/de.json]: - [ ] Like I have written you yesterday:
> Could you please do the urlencodibg programmatically. The language files are designed to be maintained by non devs and it would be pretty hard for them to write encoded text. I already commented you a snippet which should do the job.

File: locale/de.json [hunk 34]
```diff
@@ -1,4 +1,7 @@
 {
+	"contentcard.report.email": "inhalte@schul-cloud.org",
```
Comment (adrianjost) [inline locale/de.json]: - [ ] `components.organisms.` and such stuff should be part of the key. 
This was probably unclear in the documentary. We should adjust.

File: locale/en.json [hunk 35]
```diff
@@ -1,9 +1,17 @@
 {
-	"storybook.lang": "I'm in English now!",
-	"pages.news.title": "News",
-	"pages.courses.new.btn_new": "Create",
-	"pages.impressum.title": "Imprint",
+	"contentcard.report.email": "inhalte@schul-cloud.org",
+	"contentcard.report.body": "Melden%20des%20Inhaltes%20mit%20der%20ID%20&",
+	"contentcard.report.subject": "Liebes%20%20Team%2C%0Ahiermit%20m%C3%B6chte%20ich%20den%20im%20Betreff%20genannten%20Inhalt%20melden%2C%20da%3A%0A%5Bhier%20bitte%20Ihre%20Gr%C3%BCnde%5D",
```
Comment (adrianjost) [inline locale/en.json]: - [ ] same as above

File: src/components/molecules/ContentCard.unit.js [hunk 36]
```diff
@@ -0,0 +1,37 @@
+import ContentCard from "./ContentCard";
+
+describe("@components/ContentCard", () => {
+	it(...isValidComponent(ContentCard));
+
+	it("Renders content and footer slots", () => {
```
Comment (adrianjost) [inline src/components/molecules/ContentCard.unit.js]: - [ ] The  name "slots" is a bit misleading, because the test is not passing any slots in. `Renders body and footer content` may be better here.

File: src/mixins/infiniteScrolling.js [hunk 37]
```diff
@@ -0,0 +1,35 @@
...
```
Comment (adrianjost) [inline src/mixins/infiniteScrolling.js]: - [ ] I think you forgot to adjust it here as well.

File: src/pages/content/index.vue [hunk 38]
```diff
@@ -1,72 +1,113 @@
...
-		/>
-		<pagination v-model="skippedItems" :state="pagination" />
-		<div class="columns">
-			<div v-for="content of searchResults" :key="content._id" class="column">
-				<pre>
-					{{ content }}
-				</pre
-				>
+	<section>
+		<div v-if="scrollY > 115" class="content__back-to-top">
```
Comment (adrianjost) [inline src/pages/content/index.vue]: - [ ] could you please put the `115` into a variable. Magic Numbers are difficult to understand.
https://en.wikipedia.org/wiki/Magic_number_(programming)

File: locale/de.json [hunk 39]
```diff
@@ -1,4 +1,7 @@
 {
+	"components.organisms.contentcard.report.email": "inhalte@schul-cloud.org",
+	"components.organisms.contentcard.report.body": "Melden des Inhaltes mit der ID",
+	"components.organisms.contentcard.report.subject": "Liebes Team, hiermit möchte ich den im Betreff genannten Inhalt melden, da:[hier bitte Ihre Gründe]",
```
Comment (adrianjost) [inline locale/de.json]: Sorry for being unclear in my last review. I was lazy and just copied an example.
The content card is located at components/molecules. Therefore the key should start with components.molecules, not organisms.

Comment (adrianjost) [reply locale/de.json]: Update: I saw your other pull, please change it there and we will be fine.

**Comment by janrenz** [comment]
Can you update the screenshots with the new card design?

**Comment by adrianjost** [comment]
@janrenz The new design is in another pull #398
